src/officehours_api/views.py

```diff
@@ -94,6 +98,66 @@ def partial_update(self, request, *args, **kwargs):
         self.check_change_permission(request, user)
         return super().partial_update(request, *args, **kwargs)
 
+class UserOTP(DecoupledContextMixin, LoggingMixin, generics.RetrieveUpdateAPIView):
+    queryset = User.objects.all()
+    permission_classes = (IsAuthenticated,)
+    serializer_class = PhoneSerializer
+
+    def check_change_permission(self, request, user):
+        if user != request.user:
+            self.permission_denied(request)
+
+    def generate_otp(self, request):
+        request.data["otp_token"] = randint(1000, 9999)
+        request.data["otp_expiration"] = datetime.now(timezone.utc) + timedelta(minutes=5)
+
+    def verify_otp(self, request):
+        """Returns True if the OTP is correct, and a string with an error message otherwise"""
+        user = self.request.user
+        if user.profile.otp_expiration < datetime.now(timezone.utc):
+            return "Your verification code has expired. Please request a new one."
+            
+        elif user.profile.otp_token != request.data["otp_token"]:
+            return "Incorect Verification Code Entered."
+        return True
+    
+    async def send_otp(self, request, *args, **kwargs):
+        '''
+        Send the OTP to the user's phone number.
+        Returns True if the message was sent successfully, Error Response otherwise.
+        '''
+        self.generate_otp(request)
+        if await send_one_time_password(request.data["otp_phone_number"], request.data["otp_token"]):
+            return True
+        else:
+            return Response({"detail": "Failed to send verification code; please check your phone number and try again."}, 
+                            status=status.HTTP_400_BAD_REQUEST)
+    
+    def update(self, request, *args, **kwargs):
+        user = self.request.user
+        self.check_change_permission(request, user)
+
+        if request.data["action"] == "send":
```

Comment:
Can we use the twilio rest client library as shown in the sample code on the Twilio Verify site? https://www.twilio.com/docs/verify/api/verification

Reply:
I decided against Twilio Verify, as it costs [5¢ for every verification](https://www.twilio.com/en-us/verify/pricing) + standard SMS charges, and requires a separate Twilio service. 

I am unsure of what plan ITS has, but 5¢ a verification seems far too steep, so implemented it with existing SMS plan instead, which is less than 1¢ per text.

Reply:
Thanks for the thoughtfulness! Yes, we will prefer staying with the existing SMS plan. 

Could you please add the rationale either to this pull request, or into the #391 issue?